experiments/run_all_tests.rb

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require "thread"
+
+# Get all test files (excluding the windup test)
+test_files = Dir.glob("tests/*.rb").reject { |file| file.include?("test_lower_bound_windup_adaptive.rb") }.sort
+
+puts "Found #{test_files.length} test files to run:"
+test_files.each { |file| puts "  - #{file}" }
+puts "\nRunning all tests in parallel..."
+puts "=" * 60
+
+# Track results
+results = {}
+mutex = Mutex.new
+
+# Create threads for each test file
+threads = test_files.map do |test_file|
+  Thread.new do
+    start_time = Time.now
+
+    # Capture output and error
+    output = nil
+    error = nil
+    exit_status = nil
+
+    begin
+      # Run the test file and capture output
+      output = %x(bundle exec ruby #{test_file} 2>&1)
+      exit_status = $?.exitstatus
+    rescue => e
+      error = e.message
+      exit_status = 1
+    end
+
+    end_time = Time.now
+    duration = end_time - start_time
+
+    # Thread-safe result storage
+    mutex.synchronize do
+      results[test_file] = {
+        output: output,
+        error: error,
+        exit_status: exit_status,
+        duration: duration,
+      }
+
+      status = exit_status == 0 ? "✅ SUCCESS" : "❌ FAILED"
+      puts "[#{Time.now.strftime("%H:%M:%S")}] #{status} #{test_file} (#{duration.round(2)}s)"
+    end
+  end
+end
+
+# Wait for all threads to complete
+threads.each(&:join)
+
+puts "\n" + "=" * 60
+puts "All tests completed!"
+puts "=" * 60
+
+# Summary
+total_duration = results.values.map { |r| r[:duration] }.sum
+successful = results.select { |_, r| r[:exit_status] == 0 }
+failed = results.select { |_, r| r[:exit_status] != 0 }
+
+puts "\nSUMMARY:"
+puts "  Total files: #{results.length}"
+puts "  Successful: #{successful.length}"
+puts "  Failed: #{failed.length}"
+puts "  Total execution time: #{total_duration.round(2)}s"
```

Comment:
this would appear as hours even if the execution time was 10 minutes. Just because we're running things in parallel, and you're summing up the times.

Is that what you wanted to report?